Lib/pathlib.py

```diff
@@ -734,27 +643,38 @@ def is_absolute(self):
         a drive)."""
         if not self._root:
             return False
-        return not self._flavour.has_drv or bool(self._drv)
+        return self._flavour is posixpath or bool(self._drv)
```

Comment:
Is it good to hard-code a check for a specific flavour?

Reply:
I'd like to implement this via:

```python
return self._flavour.isabs(self)
```

... but `ntpath.isabs()` doesn't agree with `PureWindowsPath.is_absolute()` (see #44626) and there's no realistic prospect of fixing it, as far as I can tell.

Perhaps this would be better?

```python
# ntpath.isabs() is defective
if self._flavour is ntpath:  
    return bool(self._root and self._drv)

return self._flavour.isabs(self)
```

Reply:
UNC paths are always absolute, i.e. are never resolved against the current working directory, even if they're incomplete or invalid and have no drive, such as "//", "///", "//spam", or "//?/UNC/spam". The initial parsing could set a `_unc` attribute. Then for the `ntpath` flavour, `is_absolute()` could use  `bool(self._unc or self._root and self._drv)`.

Reply:
`_parse_parts()` already handles this:

```python
elif rel[:1] == sep or drv[:1] == sep:
    root, rel = sep, rel.lstrip(sep)
```

If the drive is a UNC path, we set the root to `'\\'`.

Reply:
Setting `root` to a single backslash has nothing to do with this. The problem is that these cases have *no drive*, but they're still absolute paths.

Reply:
Right. I'm not sure what to say except that I'm not planning to fix that in this PR. It's not a regression and it's not directly related to the aim of the PR, unless I'm still misunderstanding you.